File: credit-engine-api/src/main/java/com/credit/engine/application/service/assignor/AssignorService.java
package com.credit.engine.application.service.assignor;

import com.credit.engine.application.dto.assignor.AssignorRequest;
import com.credit.engine.application.dto.assignor.AssignorResponse;

import java.util.List;
import java.util.UUID;

public interface AssignorService {

    AssignorResponse create(AssignorRequest request);

    AssignorResponse update(UUID id, AssignorRequest request);

    AssignorResponse findById(UUID id);

    AssignorResponse findByDocumentNumber(String documentNumber);

    List<AssignorResponse> findAll();

}

File: credit-engine-ui/src/services/assignor/assignor.service.ts
import { http } from '../../api/client';
import type AssignorRequest from './dto/AssignorRequest';
import type AssignorResponse from './dto/AssignorResponse';
import { stripDocumentNumberMask } from './dto/AssignorRequest';

const controllerUrl = '/assignors';

export const assignorService = {

    findAll() {
        return http.get<AssignorResponse[]>(controllerUrl);
    },

    findByDocumentNumber(documentNumber: string) {
        const digitsOnly = stripDocumentNumberMask(documentNumber);
        return http.get<AssignorResponse>(`${controllerUrl}?documentNumber=${digitsOnly}`);
    },

    create(request: AssignorRequest) {
        return http.post<AssignorResponse>(controllerUrl, {
            ...request,
            documentNumber: stripDocumentNumberMask(request.documentNumber),
        });
    },

};
